### PR TITLE
[network] Remove address check on LostPeer notifs

### PR DESCRIPTION
**Pull request by gregnazario**
During LostPeer notifications, the address may mismatch what is in
storage.  This removes that check and will disconnect the peer if it
doesn't match, and we'll let us later redial or otherwise handle the
connection.

Related to: https://github.com/libra/libra/issues/5182

### CODE REVIEW

File: network/src/connectivity_manager/mod.rs [hunk 1]
```diff
@@ -118,6 +118,9 @@ pub enum ConnectivityRequest {
     UpdateEligibleNodes(DiscoverySource, HashMap<PeerId, HashSet<x25519::PublicKey>>),
     /// Gets current size of dial queue. This is useful in tests.
     #[serde(skip)]
+    GetConnectedSize(oneshot::Sender<usize>),
```
Comment (phlip9) [inline network/src/connectivity_manager/mod.rs:121]: nit: comment is copy-pasted : p

File: network/src/connectivity_manager/mod.rs [hunk 2]
```diff
@@ -597,19 +603,16 @@ where
...
-                        debug!(
-                            "{} Ignoring stale lost peer event for peer: {}, addr: {}",
-                            self.network_context,
-                            peer_id.short_str(),
-                            addr
-                        );
-                    }
+                if self.connected.contains_key(&peer_id) {
+                    // Remove node from connected peers list.
+                    self.connected.remove(&peer_id);
```
Comment (davidiw) [inline network/src/connectivity_manager/mod.rs:608]: do we want to log this as well?

Comment (gregnazario) [reply network/src/connectivity_manager/mod.rs:608]: Luckily now we have the incoming / outgoing messages from the events.

I'll add a structured log for every event received

File: network/src/connectivity_manager/mod.rs (hunk 1 above)
Comment (phlip9) [inline network/src/connectivity_manager/mod.rs:121]: man, i wish we could refactor these actors so we didn't need these test-only messages...

Comment (gregnazario) [reply network/src/connectivity_manager/mod.rs:121]: I actually tried to refactor it, and then gave up to do this instead

Comment (gregnazario) [reply network/src/connectivity_manager/mod.rs:121]: :(

File: network/src/connectivity_manager/mod.rs (hunk 2 above)
Comment (gregnazario) [reply network/src/connectivity_manager/mod.rs:608]: Oh, turns out we already log the entire event in the structured logs event loop, with the internals, which would tell the directions and all.

Comment (davidiw) [reply network/src/connectivity_manager/mod.rs:615]: Agreed! But that doesn't show up per our previous conversation. Once that works, I think we can probably start eliminating a bunch of logs. I don't know if there's any value from having both structured and text logs. We eliminated text logs from the TCB unless there was something that didn't fit into the model for structured loggings.

**Pull request closed without merging**